core/templates/dev/head/pages/exploration_player/PlayerServices.js

```diff
@@ -213,6 +213,11 @@ oppia.factory('oppiaPlayerService', [
                 _explorationId));
           }
           loadedExploration.then(function(data) {
+            // Update exploration's states field.
+            for (var stateName in data.exploration.states) {
```

Comment:
Ah, sorry, no. That still won't do. It's important to not modify the keys of objects/sub-objects, otherwise their definition becomes quite unclear.

Here's an alternative. If this is only used in the learner view, consider having a separate StateToClassifierMappingService that is only available in the learner view. The Exploration and State domain objects remain as they are. Just fetch the classifier data from this service as needed.